src/sinks/axiom.rs

```diff
@@ -0,0 +1,400 @@
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    acknowledgements: AcknowledgementsConfig,
+}
+
+inventory::submit! {
+    SinkDescription::new::<AxiomConfig>("axiom")
+}
+
+impl GenerateConfig for AxiomConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(
+            r#"token = "${AXIOM_TOKEN}"
+            dataset = "my-dataset"
+            url = "${AXIOM_URL}"
+            org_id = "${AXIOM_ORG_ID}""#,
+        )
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "axiom")]
+impl SinkConfig for AxiomConfig {
+    async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
+        let mut headers = IndexMap::with_capacity(1);
+        headers.insert(
+            "X-Axiom-Org-Id".to_string(),
+            self.org_id.clone().unwrap_or_default(),
+        );
+        let mut query = HashMap::with_capacity(1);
+        query.insert("timestamp-field".to_string(), "timestamp".to_string());
+        let elasticsearch_config = ElasticsearchConfig {
+            endpoint: self.build_endpoint(),
+            compression: self.compression,
+            auth: Some(ElasticsearchAuth::Basic {
+                user: "axiom".to_string(),
+                password: self.token.clone(),
+            }),
+            query: Some(query),
+            tls: self.tls.clone(),
+            request: RequestConfig {
+                headers,
+                ..Default::default()
+            },
+            ..Default::default()
+        };
+        let config = elasticsearch_config.build(cx).await?;
+        Ok(config)
+    }
+
+    fn input(&self) -> Input {
+        Input::log()
+    }
+
+    fn sink_type(&self) -> &'static str {
+        "axiom"
+    }
+
+    fn acknowledgements(&self) -> Option<&AcknowledgementsConfig> {
+        Some(&self.acknowledgements)
+    }
+}
+
+impl AxiomConfig {
+    fn build_endpoint(&self) -> String {
+        let url = if let Some(url) = self.url.as_ref() {
+            url.clone()
+        } else {
+            CLOUD_URL.to_string()
+        };
+
+        format!("{}/api/v1/datasets/{}/elastic", url, self.dataset)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    #[test]
+    fn generate_config() {
+        crate::test_util::test_generate_config::<super::AxiomConfig>();
+    }
+}
+
+#[cfg(feature = "axiom-integration-tests")]
+#[cfg(test)]
+mod integration_tests {
+    use chrono::{DateTime, Duration, Utc};
+    use futures::stream;
+    use http::StatusCode;
+    use std::env;
+    use tokio::time;
+    use vector_core::event::{BatchNotifier, BatchStatus, Event, LogEvent};
+
+    use super::*;
+    use crate::{
+        config::SinkContext,
+        sinks::axiom::AxiomConfig,
+        test_util::{
+            components::{run_and_assert_sink_compliance, HTTP_SINK_TAGS},
+            wait_for_duration,
+        },
+    };
+
+    #[tokio::test]
+    async fn axiom_logs_put_data() {
```

Comment:
👍 I could see pulling out the setup bits, but since we only have one test it's fine in here for now.

Reply:
Agree, if we start re-using it pulling it out makes sense but I think it's fine as-is today.